examples/showcase_1kt.rs

```diff
@@ -0,0 +1,240 @@
+
+        self.offset
+    }
+
+    fn inner(&self) -> &[u8] {
+        &self.buf
+    }
+}
+
+fn main() {
+    // Configure the radio
+    let pan: u16 = 0xcafe;
+    let addr_short: u16 = 0xdead;
+    let addr_long: u64 = 0xdead_bead;
+    let tx_power: i8 = 5;
+    let channel: u8 = 11;
+
+    Ieee802154::set_pan(pan);
+    Ieee802154::set_address_short(addr_short);
+    Ieee802154::set_address_long(addr_long);
+    Ieee802154::set_tx_power(tx_power).unwrap();
+    Ieee802154::set_channel(channel).unwrap();
+
+    // Don't forget to commit the config!
+    Ieee802154::commit_config();
+
+    // Turn the radio on
+    Ieee802154::radio_on().unwrap();
+    assert!(Ieee802154::is_on());
+
+    // TODO: think how to make Alarm a singleton.
+    // Multiple Alarms would clash.
+    let mut alarm = Alarm::new();
+
+    let mut frames_buf1 = RxRingBuffer::<N_MOTES>::new();
+    let mut frames_buf2 = RxRingBuffer::<N_MOTES>::new();
+    let mut operator =
+        RxBufferAlternatingOperator::new(&mut frames_buf1, &mut frames_buf2).unwrap();
+
+    let mut sequence_no = 0_usize;
+    let mut msg_buf = MsgBuf::<MSG_BUF_LEN>::new();
+
+    let cherry_id = get_cherry_id()
+        .map(str::parse::<u32>)
+        .transpose()
+        .unwrap()
+        .unwrap_or(0);
+
+    loop {
+        // Measure temperature.
+        let temperature_centigrades_celcius = Temperature::read_temperature_sync().unwrap();
+
+        // Get current time.
+        let timestamp = Alarm::get_ticks().unwrap();
+
+        // Fill the buffer with current data.
+        let msg_len = msg_buf.fill(timestamp, sequence_no, temperature_centigrades_celcius);
+
+        // Transmit a frame
+        Ieee802154::transmit_frame(&msg_buf.inner()[..msg_len]).unwrap();
+
+        writeln!(
+            ConsoleLite::writer(),
+            "Transmitted frame {} of len {}!\n",
+            sequence_no,
+            msg_len,
+        )
+        .unwrap();
+
+        // Sleep for a predefined period of time, so that each mote sends its message.
+        Alarm::sleep_for(SLEEP_TIME).unwrap();
+
+        // For each peer...
+        for _ in 0..N_MOTES - 1 {
+            // Receive a frame from it.
+            let maybe_frame = operator
+                .receive_frame_timed(&mut alarm, Milliseconds(10))
+                .unwrap();
+
+            struct Ascii<'a>(&'a [u8]);
+            impl Display for Ascii<'_> {
+                fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                    for b in self.0.iter().copied() {
+                        let c = char::from_u32(b as u32).unwrap_or('*');
+                        f.write_char(c)?;
+                    }
+                    Ok(())
+                }
+            }
+
+            if let Some(frame) = maybe_frame {
+                let body_len = frame.payload_len as usize;
+                let raw_body = &frame.body[..body_len];
+                let decoded_frame = core::str::from_utf8(raw_body);
```

Comment:
To wygląda ,,ładnie'' w Rust :)
Docelowo liczę, że jak będzie wiadomość-struct, to będzie podobnie eleganckie from/into.